src/runtime/runtime_rpi3.go

```diff
@@ -0,0 +1,140 @@
+// +build rpi3
+
+package runtime
+
+import (
+	"machine"
+	"runtime/volatile"
+	"unsafe"
+	"device/arm"
+)
+
+const tickMicros = int64(1)
+
+type timeUnit int64
+
+//go:export sleepticks sleepticks
+func sleepTicks(n timeUnit) {
+	WaitMuSec(uint32(n))
+}
+
+func ticks() timeUnit {
+	return timeUnit(SysTimer())
+}
+
+func preinit() {
+	//XXX because interp.Unsupported would be thrown if we tried to get this to happen at initializer time
+	machine.Aux = (*machine.AuxPeripheralsRegisterMap)(unsafe.Pointer(machine.MemoryMappedIO + 0x00215000))
+	machine.GPIO = (*machine.GPIORegisterMap)(unsafe.Pointer(machine.MemoryMappedIO + 0x00200000))
+	machine.SysTimer = (*machine.SysTimerRegisterMap)(unsafe.Pointer(machine.MemoryMappedIO + 0x3000))
+
+
+
+	// Initialize .bss: zero-initialized global variables.
+	//ptr := unsafe.Pointer(&_sbss)
+	//for ptr != unsafe.Pointer(&_ebss) {
+	//	*(*uint32)(ptr) = 0
+	//	ptr = unsafe.Pointer(uintptr(ptr) + 4)
+	//}
+
+	machine.UARTInit()
+}
+
+// main on bare metal (bootloader uses this path)
+//go:export main
+func main() {
+	preinit()
+	primaryMain()
```

Comment:
If you rebase on `dev`, there is a convenient wrapper called `run()` which you can use to replace this. Then your code could be changed to:
```
preinit()
run()
machine.Abort()
```

If the scheduler does not work yet, you can set the scheduler to `none` in the device file, like we currently do for avr: https://github.com/tinygo-org/tinygo/blob/cbaa58a2d9f4c29a2f0fbd79130fee62b8a1f72a/targets/avr.json#L8

Reply:
Yep, done.